FluentUI/src/main/java/com/microsoft/fluentui/contextualcommandbar/ContextualCommandBar.kt

```diff
@@ -0,0 +1,262 @@
+            gravity = Gravity.END
+        }
+
+        // Dismiss gradient gap
+        val gradientGap = View(context).apply {
+            background = ContextCompat.getDrawable(
+                    context,
+                    R.drawable.contextual_command_bar_dismiss_button_gap_background
+            )
+        }
+        dismissButtonContainer!!.addView(gradientGap)
+        gradientGap.layoutParams.apply {
+            width = resources.getDimensionPixelSize(R.dimen.fluentui_contextual_command_bar_dismiss_gap_width)
+            height = ViewGroup.LayoutParams.MATCH_PARENT
+        }
+
+        // Dismiss button
+        dismissButton = ImageView(context).apply {
+            setPaddingRelative(
+                    itemPaddingHorizontal,
+                    itemPaddingVertical,
+                    itemPaddingHorizontal,
+                    itemPaddingVertical
+            )
+            setBackgroundColor(ThemeUtil.getColor(
+                    context,
+                    R.attr.fluentuiContextualCommandBarDismissBackgroundColor
+            ))
+            imageTintList = ColorStateList.valueOf(ThemeUtil.getColor(
+                    context,
+                    R.attr.fluentuiContextualCommandBarDismissIconTintColor
+            ))
+
+            setOnClickListener {
+                dismissListener?.invoke()
+            }
+        }
+        dismissButtonContainer!!.addView(dismissButton)
+        dismissButton!!.layoutParams.apply {
+            height = ViewGroup.LayoutParams.MATCH_PARENT
+            width = resources.getDimensionPixelSize(
+                    R.dimen.fluentui_contextual_command_bar_dismiss_button_width
+            )
+        }
+
+        bringChildToFront(dismissButtonContainer)
+        setDismissButtonVisible(showDismiss)
+    }
+
+    private fun setDismissButtonVisible(visible: Boolean) {
+        val paddingEnd = if (visible) {
+            resources.getDimensionPixelSize(
+                    R.dimen.fluentui_contextual_command_bar_dismiss_button_width
+            ) + resources.getDimensionPixelSize(
+                    R.dimen.fluentui_contextual_command_bar_dismiss_gap_width
+            )
+        } else 0
+        commandContainer.setPaddingRelative(0, 0, paddingEnd, 0)
+        dismissButtonContainer?.isVisible = visible
+    }
+
+    fun setItemGroups(itemGroups: List<CommandItemGroup>) {
+        commandContainer.removeAllViews()
+
+        for ((groupIdx, itemGroup) in itemGroups.withIndex()) {
+            val items = itemGroup.items
+            if (items.isEmpty()) {
+                continue
+            }
+
+            val groupContainer = LinearLayout(context)
+            commandContainer.addView(groupContainer)
+            if (groupIdx != itemGroups.size - 1) {
+                (groupContainer.layoutParams as LinearLayout.LayoutParams).apply {
+                    marginEnd = groupSpace
+                }
+            }
+
+            for ((idx, item) in items.withIndex()) {
```

Comment:
can we break it to another function  for more readability 
like addItemViewToGroup or similar